File: mycuration/src/main/java/com/phicdy/mycuration/tracker/GATrackerHelper.java
package com.phicdy.mycuration.tracker;

import com.google.android.gms.analytics.HitBuilders;
import com.google.android.gms.analytics.Tracker;
import com.phicdy.mycuration.BuildConfig;

public class GATrackerHelper {

    private static Tracker tracker;
    private static String categoryAction;

    public static void setTracker(Tracker newTracker) {
        tracker = newTracker;
    }

    public static void setCategoryAction(String action) {
        categoryAction = action;
    }

    public static void sendScreen(String screenName) {
        if (BuildConfig.DEBUG) return;
        if (tracker == null) throw new IllegalStateException();
        tracker.setScreenName(screenName);
        tracker.send(new HitBuilders.ScreenViewBuilder().build());
    }

    public static void sendEvent(String action) {
        if (BuildConfig.DEBUG) return;
        if (tracker == null) throw new IllegalStateException();
        tracker.send(new HitBuilders.EventBuilder()
                .setCategory(categoryAction)
                .setAction(action)
                .build());
    }

    public static void sendEvent(String action, long value) {
        if (BuildConfig.DEBUG) return;
        if (tracker == null) throw new IllegalStateException();
        tracker.send(new HitBuilders.EventBuilder()
                .setCategory(categoryAction)
                .setAction(action)
                .setValue(value)
                .build());
    }
}

File: mycuration/src/main/java/com/phicdy/mycuration/tracker/GATrackerHelper.kt
package com.phicdy.mycuration.tracker

import com.google.android.gms.analytics.HitBuilders
import com.google.android.gms.analytics.Tracker
import com.phicdy.mycuration.BuildConfig

object GATrackerHelper {

    private lateinit var tracker: Tracker
    private lateinit var categoryAction: String

    fun setTracker(newTracker: Tracker) {
        tracker = newTracker
    }

    fun setCategoryAction(action: String) {
        categoryAction = action
    }

    fun sendScreen(screenName: String) {
        if (BuildConfig.DEBUG) return
        tracker.setScreenName(screenName)
        tracker.send(HitBuilders.ScreenViewBuilder().build())
    }

    fun sendEvent(action: String) {
        if (BuildConfig.DEBUG) return
        tracker.send(HitBuilders.EventBuilder()
                .setCategory(categoryAction)
                .setAction(action)
                .build())
    }

    fun sendEvent(action: String, label: String) {
        if (BuildConfig.DEBUG) return
        tracker.send(HitBuilders.EventBuilder()
                .setCategory(categoryAction)
                .setAction(action)
                .setLabel(label)
                .build())
    }
}
